### PR TITLE
Implement connection protocol in python agent

### PR DESCRIPTION
**Pull request by dbluhm**
All credit to @trthhrtz for his excellent work implementing the connection protocol as it currently stands in the connection protocol HIPE inside of the python agent.

### CODE REVIEW

File: python/message_types.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+class CONN:
+    """ Connetion Class of Message Types.
+
+        This type notation is still being discussed and may change.
+    """
+
+
+class UI:
+    STATE = "urn:sovrin:agent:message_type:sovrin.org/ui/state"
```
Comment (TelegramSam) [inline python/message_types.py]: These message type strings are not consistent with the Message Types Indy HIPE. Should this be updated now, or is it using this older style string for compatibility?

Comment (dbluhm) [reply python/message_types.py]: Good catch. I think we ought to update this now. I can't think of any reason to keep it around for compatibility purposes. I'll bring this up to @trthhrtz.

File: python/modules/connection.py [hunk 2]
```diff
@@ -4,220 +4,222 @@
...
     )
 
-    (my_did, _) = await did.create_and_store_my_did(wallet_handle, "{}")
-
-    await did.store_their_did(wallet_handle, ident_json)
+    await did.set_did_metadata(agent.wallet_handle, endpoint_did_str, meta_json)
 
-    await did.set_endpoint_for_did(wallet_handle, accept_did, endpoint, verkey)
+    inner_msg = Message(
+        type= "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/connections/1.0/request",
```
Comment (TelegramSam) [inline python/modules/connection.py]: For consistency, these types should be referenced in the same way the other type names have been.

File: python/README.md [hunk 3]
```diff
@@ -4,3 +4,30 @@ indy-agent implementation in Python
...
+- Python 3.6
+- Latest `libindy` from https://github.com/hyperledger/indy-sdk
+
+Quickstart
+----------
+
+First, create a virtual environment to install indy-agent dependencies:
+
+```sh
+cd python/
```
Comment (TelegramSam) [inline python/README.md]: This needs a little more depth. which path are we expected to be in here? Obviously we won't know the full path, but we can reference something like where the code was checked out, or some other path the user has created for a specific purpose.

**Comment by dbluhm** [comment]
@TelegramSam merged Kuzma's changes.

**Pull request merged**